firefox-ios/Client/Frontend/Home/Homepage Rebuild/TopSites/TopSitesDimensionImplementation.swift

```diff
@@ -17,6 +40,14 @@ struct TopSitesDimensionImplementation {
             availableWidth = availableWidth - cellWidth - HomepageSectionLayoutProvider.UX.standardSpacing
         }
         let minCardsConstant = HomepageSectionLayoutProvider.UX.TopSitesConstants.minCards
-        return numberOfTiles < minCardsConstant ? minCardsConstant : numberOfTiles
+        let tilesPerRowCount = numberOfTiles < minCardsConstant ? minCardsConstant : numberOfTiles
+
+        // Since this method gets called several multiple times, including when we update top sites state
+        // we add the following to ensure that the call to update numberOfTilesPerRow completes
+        queue.async {
+            self.numberOfTilesPerRow = tilesPerRowCount
+        }
```

Comment:
I still am a bit confused about why we need this dispatch to the main thread asynchronously. Regardless of whether we keep this functionality could we update the comment with the follow up ticket number and to be a bit more descriptive? Is "add the following to ensure that the call to update numberOfTilesPerRow completes" an accurate description of this code? Or is it more we add the following to ensure that `numberOfTilesPerRow` is updated on the main thread or something?

Reply:
thanks @Cramsden - I added the ticket here: https://mozilla-hub.atlassian.net/browse/FXIOS-10972